src/tooltip/ToolTip2.tsx

```diff
@@ -0,0 +1,235 @@
+import React from 'react';
+import {
+  Modal,
+  View,
+  I18nManager,
+  Pressable,
+  ViewStyle,
+  FlexStyle,
+  StyleProp,
+  StyleSheet,
+  ColorValue,
+  TouchableWithoutFeedback,
+} from 'react-native';
+import { Animated, Text, SafeAreaView } from 'react-native';
+import Triangle from './Triangle';
+import { withTheme } from '../config';
+import { ScreenWidth, ScreenHeight } from '../helpers';
+
+export type ToolTip2Props = {
+  withPointer?: boolean;
+  title?: string;
+  popover?: React.ReactElement<{}>;
+  toggleAction?: 'onPress' | 'onLongPress';
+  width?: FlexStyle['width'];
+  containerStyle?: StyleProp<ViewStyle>;
+  pointerColor?: ColorValue;
+  onDismiss?(): void;
+  onShow?(): void;
+  visible?: boolean;
+  overlayColor?: ColorValue;
+  backgroundColor?: ColorValue;
+  closeOnlyOnBackdropPress?: boolean;
+};
+
+const ToolTip2: React.FunctionComponent<ToolTip2Props> = ({
+  children,
+  title,
+  popover,
+  withPointer = true,
+  overlayColor,
+  backgroundColor = '#616161',
+  pointerColor,
+  toggleAction = 'onPress',
+  onShow = () => {},
+  onDismiss = () => {},
+  visible = false,
+  width: propWidth = 200,
+  closeOnlyOnBackdropPress = false,
+  containerStyle,
+}) => {
+  const { current: animation } = React.useRef<Animated.Value>(
+    new Animated.Value(0)
+  );
+
+  let containerElement = React.useRef<View>(null);
+  let tooltipElement = React.useRef<View>(null);
+  const [containerDimensions, setContainerDimensions] = React.useState({
+    height: 0,
+    px: 0,
+    py: 0,
+    width: 0,
+  });
+  const [tooltipDimension, setTooltipDimension] = React.useState({
+    height: 0,
+    width: 0,
+  });
+
+  const toogleToolTip = () => {
+    if (visible) {
+      onDismiss();
+    } else {
+      onShow();
+    }
+  };
+
+  React.useEffect(() => {
+    Animated.timing(animation, {
+      toValue: Number(visible),
+      useNativeDriver: true,
+      duration: 200,
+    }).start();
+  }, [visible, animation]);
+
+  React.useEffect(() => {
+    let subs = true;
```

Comment:
Asking out of curiosity what's that `subs` used for? Causing error when you change state too quickly or something?

Reply:
This will not allow to update the state if component gets unmounted